__tests__/app.test.js

```diff
@@ -98,6 +71,80 @@ describe('GET /api/reviews/:review_id', () => {
           expect(res.body.msg).toBe("bad request");
         });
     });
-  });
 
 
+
+    describe('PATCH /api/reviews/:review_id', () => {
+        test('200:return the updated vote in the response body', () => {
+          const incrementedVoteObj = { inc_votes: 100 };
+          return request(app)
+            .patch('/api/reviews/1')
+            .send(incrementedVoteObj)
+            .expect(200)
+            .then(({ body }) => {
+                console.log(body);
+              expect(body.review.review_id).toBe(1);
+              expect(body.review.votes).toBe(101);
+            });  
+        });
+
+        test("200: returns returns updated review object when new votes is a negative number", () => {
+            const newReqBody = { inc_votes: -2 };
+            return request(app)
+              .patch("/api/reviews/2")
+              .send(newReqBody)
+              .expect(200)
+              .then((res) => {
+                expect(res.body.review).toEqual({
+                  review_id: 2,
+                  title: "Jenga",
+                  designer: "Leslie Scott",
+                  owner: "philippaclaire9",
+                  review_img_url:
+                    "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+                  review_body: "Fiddly fun for all the family",
+                  category: "dexterity",
+                  created_at: "2021-01-18T10:01:41.251Z",
+                  votes: 3,
+                });
+              });
+          });
+
+        test('404: when provided with a valid but non-existent review id', () => {
+            const newVoteObj = { inc_votes: 1 };
+            return request(app)
+              .get('/api/reviews/1000')
+              .send(newVoteObj)
+              .expect(404)
+              .then(({ body }) => {
+                expect(body.msg).toBe('not found');
+              });
+          });
+
+          test("400: returns bad request error message when called with a review id of wrong data type", () => {
+            const reqBody = { inc_votes: 7 };
+        
+            return request(app)
+              .patch("/api/reviews/bananas")
+              .send(reqBody)
+              .expect(400)
+              .then((res) => {
+                expect(res.body.msg).toBe("bad request");
+              });
+          });
+
+          test("400: returns an error message when called with a float/decimal number ", () => {
+            const newReqBody = { inc_votes: "5.5" };
+        
+            return request(app)
+              .patch("/api/reviews/1")
+              .send(newReqBody)
+              .expect(400)
+              .then((res) => {
+                expect(res.body.msg).toBe("bad request");
+              });
+          });
+        });
+     });
+    //});
+//});
```

Comment:
I would like to see a test with a broken/missing inc_votes key such as sending {} or {votes: 100}

Reply:
Also not a problem really but i would like you to be more consistent with your naming. In this describe block you have used newReqBody, reqBody, newVoteObj, newReqBody, and incrementVoteObj, to refer to the same thing in different tests. Any variables you write in the test block are scoped to that test function so we dont have to worry about name conflicts and it is much better for readability to have these things named the same so we can tell they are serving the same purpose, just with different data.

Reply:
https://github.com/gsbinning/nc_games/pull/4/commits/1c731be5bc40cd6590bfdd040835b8c304712f3b